src/Input/Input.js

```diff
@@ -369,8 +402,16 @@ class Input extends Component<DefaultProps, AllProps, State> {
     }
   };
 
+  /**
+   * A controlled input accepts its current value as a prop, as well as a callback
+   * to change that value.
+   *
+   * @returns {boolean}
+   * @see https://facebook.github.io/react/docs/forms.html#controlled-components
+   */
   isControlled() {
-    return typeof this.props.value === 'string';
+    const { onChange, value } = this.props;
```

Comment:
onChange shouldn't be taken into account. React isn't using it for native components.